Lib/ufo2ft/instructionCompiler.py

```diff
@@ -0,0 +1,310 @@
+from __future__ import annotations
+
+import array
+import logging
+from typing import TYPE_CHECKING, Optional
+
+from fontTools import ttLib
+from fontTools.pens.hashPointPen import HashPointPen
+from fontTools.ttLib import newTable
+from fontTools.ttLib.tables._g_l_y_f import (
+    OVERLAP_COMPOUND,
+    ROUND_XY_TO_GRID,
+    USE_MY_METRICS,
+)
+
+from ufo2ft.constants import (
+    OBJECT_LIBS_KEY,
+    TRUETYPE_INSTRUCTIONS_KEY,
+    TRUETYPE_METRICS_KEY,
+    TRUETYPE_OVERLAP_KEY,
+    TRUETYPE_ROUND_KEY,
+)
+
+if TYPE_CHECKING:
+    from fontTools.ttLib.tables._g_l_y_f import Glyph as TTGlyph
+    from ufoLib2 import Font, Glyph
+
+
+logger = logging.getLogger(__name__)
+
+
+class InstructionCompiler:
+    def __init__(
+        self, ufo: Font, otf: ttLib.TTFont, autoUseMyMetrics: bool = True
+    ) -> None:
+        self.ufo = ufo
+        self.otf = otf
+        if not autoUseMyMetrics:
+            # If autoUseMyMetrics is False, replace the method with a no-op
+            self.autoUseMyMetrics = lambda *args: None
```

Comment:
```suggestion
            self.autoUseMyMetrics = lambda ttGlyph, glyphName: None
```

Otherwise, this would change the function signature.